File: Leetcode/2044 - Count Number of Maximum Bitwise-OR Subsets/solution.py
def countMaxOrSubsets(self, nums: List[int]) -> int:
    def getMaxORSubsetCount(nums: List[int], index: int, maxOR: int, currOR: int) -> int:
        if index == len(nums):
            return 1 if maxOR == currOR else 0
        else:
            include = getMaxORSubsetCount(nums, index + 1, maxOR, currOR | nums[index])
            exclude = getMaxORSubsetCount(nums, index + 1, maxOR, currOR)
            return include + exclude

    maxOR = 0
    for num in nums:
        maxOR |= num
    return getMaxORSubsetCount(nums, 0, maxOR, 0)

File: Leetcode/2044 - Count Number of Maximum Bitwise-OR Subsets/solution.cpp
int getMaxORSubsetCount(vector<int>& nums, int index, int& maxOR, int currOR) {
    if (index == nums.size()) {
        return (currOR == maxOR) ? 1 : 0;
    }

    int include = getMaxORSubsetCount(nums, index + 1, maxOR, currOR | nums[index]);
    int exclude = getMaxORSubsetCount(nums, index + 1, maxOR, currOR);

    return include + exclude;
}

int countMaxOrSubsets(vector<int>& nums) {
    int maxOR = 0;
    for (int num : nums) {
        maxOR |= num;
    }

    return getMaxORSubsetCount(nums, 0, maxOR, 0);
}